### PR TITLE
Added TableSummaryView for inline tables

### PR DESCRIPTION
**Pull request by sarthak-marwaha**
- Added a TableSummaryView for inline tables that shows a limited number of entries and an option to expand to a fullscreen table
- Memento Fixes for TabbedDetailView
- Follow up work recorded in https://nanopay.atlassian.net/browse/NP-4944

image

### CODE REVIEW

File: src/foam/u2/stack/BreadcrumbView.js [hunk 1]
```diff
@@ -29,27 +35,29 @@ foam.CLASS({
...
       this.addClass(this.myClass('display'));
-      var navStack = this.stack.stack_.slice(this.stack.navStackBottom, this.stack.pos);
-      var themeIcon = navStack.length == 1 ? 'back' : '';
-      navStack.map((v, i, _) =>{
-        var index = i + this.stack.navStackBottom;
-        var jumpAction  = self.Action.create({
-          name: 'back',
-          code: () => {
-            self.stack.jump(index, self);
+      if ( this.stack && this.stack.stack_ ) {
```
Comment (sarthak-marwaha) [inline src/foam/u2/stack/BreadcrumbView.js:38]: Not sure why git is reading this as a massive change but only this condition was added in this file

File: src/foam/u2/view/TableSummaryView.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
+/**
+ * @license
+ * Copyright 2017 The FOAM Authors. All Rights Reserved.
```
Comment (kgrgreer) [inline src/foam/u2/view/TableSummaryView.js]: 2021

File: src/foam/u2/view/TableSummaryView.js [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.u2.view',
+  name: 'TableSummaryView',
```
Comment (kgrgreer) [inline src/foam/u2/view/TableSummaryView.js]: SummaryView has another meaning. It is a synonym of CitationView. Can you think of a better name?

Comment (jlhughes) [reply src/foam/u2/view/TableSummaryView.js]: EmbeddedTableView, InlineTableView ?

Comment (sarthak-marwaha) [reply src/foam/u2/view/TableSummaryView.js]: I like EmbeddedTableView, InlineTableView is too similar to InlineBrowserView

**Comment by kgrgreer** [comment]
Has a merge conflict.

**Pull request merged**